commons/Common.mli

```diff
@@ -140,6 +140,44 @@ val write_file : file:filename -> string -> unit
 *)
 val read_file : ?max_len:int -> filename -> string
 
+val is_directory : filename -> bool
+
+
+type path_special = private Path of string [@@deriving show, eq]
```

Comment:
some comments about private would be useful (given it's an advanced OCaml feature people may not know about).
You can copy what you had in our PR subject:
"private means that it can be matched upon to extract the constituent value (just a string), but it cannot be freely used to construct values of the type. Instead, they must use the Path.of_string function, which normalizes the path. This makes sure that all values of type Path.t must be completely normalized paths."